src/lib/storage/dictionary_segment/attribute_vector_iterable.hpp

```diff
@@ -88,17 +91,19 @@ class AttributeVectorIterable : public PointAccessibleSegmentIterable<AttributeV
     ChunkOffset _chunk_offset;
   };
 
-  template <typename ZsDecompressorType>
+  template <typename ZsDecompressorType, typename _PosListIteratorType>
   class PointAccessIterator
-      : public BasePointAccessSegmentIterator<PointAccessIterator<ZsDecompressorType>, SegmentPosition<ValueID>> {
+      : public BasePointAccessSegmentIterator<PointAccessIterator<ZsDecompressorType, _PosListIteratorType>,
+                                              SegmentPosition<ValueID>, _PosListIteratorType> {
    public:
+    using PosListIteratorType = _PosListIteratorType;
```

Comment:
This was an attempt to only pass the PosListIteratorType to base class templates once. Note how the BasePointAccessSegmentIterator is given `_PosListIterator` twice, once indirectly in the first template parameter (which is the current class) and once as the last parameter. We thought that if we made the template parameter publicly available in the current class, we wouldn't have to add that last parameter, because the BasePointAccessSegmentIterator could just use `Derived::PosListIterator`.  However, it didn't work, so currently, we still pass it twice all the time.

Do you knowhow to correctly expose and use the template parameter in the super class?
Otherwise, I guess we will remove all the leading underscores and using directives.

Reply:
Why is passing the parameter twice an issue?

Reply:
We considered having two parameters that will always be given the same value a code smell.

I'll remove all the underscores and `using PolListIterator = _PosListIterator` statements.

Reply:
Sounds good. I'm working on other PRs in parallel. In case you have any questions, we can also set up an ad-hoc call. Maybe we can get this done in the next hours?

Reply:
(Please send me a mail if you think that makes sense. Github notifications seem to be a bit delayed.)

Reply:
> I'll remove all the underscores and `using PolListIterator = _PosListIterator` statements.

done.

Thanks for being so responsive :)